spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/stream/SkipperStreamDeployer.java

```diff
@@ -289,15 +292,23 @@ private void validateAllAppsRegistered(StreamDeploymentRequest streamDeploymentR
 			// nothing to validate.
 			return;
 		}
-
+		String streamName = streamDeploymentRequest.getStreamName();
 		// throw as at this point we should have definition
 		StreamDefinition streamDefinition = this.streamDefinitionRepository
-				.findById(streamDeploymentRequest.getStreamName())
+				.findById(streamName)
 				.orElseThrow(() -> new NoSuchStreamDefinitionException(streamDeploymentRequest.getStreamName()));
 
 		for (AppDeploymentRequest adr : streamDeploymentRequest.getAppDeploymentRequests()) {
+			String registeredAppName = getRegisteredName(streamDefinition, adr.getDefinition().getName());
+			String appName =  String.format("%s-%s-v", streamName, registeredAppName);
+			// Give 2 digits for version for now
```

Comment:
Never really though about how high that number can go before.  We don't want to end up in a situation where we can get to version 99 and then not to 100.  How about we leave 4 or 5 digits (famous last words) and also log a warning if the stream name itself is over 20 chars and the registered app name is over 20 chars... no real reason for them to be super long.

We can also do a check in skipper once we know the final version number.